app/gui2/src/providers/widgetTree.ts

```diff
@@ -5,6 +5,20 @@ import { Ast } from '@/util/ast'
 import type { Icon } from '@/util/iconName'
 import { computed, proxyRefs, type Ref } from 'vue'
 
+function makeExistenceRegistry(onChange: (anyExist: boolean) => void) {
```

Comment:
The function returning lambda returning lambda without any names is hard to understands.

Let's instead return object with "register" function with documentation, that "register" returns "unregister" handle... or let it be again an object with "unregister" field.